### PR TITLE
MES-88 - Test report timer

### PR DESCRIPTION
**Pull request by Cavalier07**
## Description and relevant Jira numbers

Adds a timer to the test report
User can start, pause and restart the timer by tapping on it. Screenshots below.

## Pull Request checklist:

- [x] [WIP] tag removed from PR title
- [x] PR has an understandable description

## Git feature branch checklist:

- [x] branch name comply with our branching strategy
- [x] git branch contains relevant JIRA ticket number
- [x] branch rebased against the latest develop
- [x] commits are squashed

## Sign off process checklist:

- [x] Code has been tested manually
- [x] Tests are passing
- [x] PR link added to JIRA
- [x] Reviewers selected in Github
- [x] Any new reusable component/widget added to component library on Confluence

- [x] Screenshot(s) captured on the physical device (if applicable)
- [ ] Tested by QA
- [x] PO's approval

**Default state**
IMG_686F30388C75-1

**Paused state**
IMG_F765226877E6-1

**Active state**
IMG_E31A52DBB4CB-1

### CODE REVIEW

File: src/pages/test-report/components/timer/timer.ts [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+  generateTimerString = (): void => {
+    const date: Date = new Date(1970, 0, 1 , 0 , 0, 0);
+    date.setSeconds(this.seconds);
+
+    const showExtraZeroHours = date.getHours() < 10;
+    const showExtraZeroMinutes = date.getMinutes() < 10;
+    const showExtraZeroSeconds = date.getSeconds() < 10;
+
+    this.timerString = `${
```
Comment (johnmorleybjss) [inline src/pages/test-report/components/timer/timer.ts]: Could we use use our `DateTime` provider to do some of the heavy lifting for us here?

Comment (ammarhaiderbjss) [reply src/pages/test-report/components/timer/timer.ts]: Could be, but I kinda like that it's a self-contained timer

Comment (buzzub) [reply src/pages/test-report/components/timer/timer.ts]: Nice and readable. I'd have been tempted to do something like `(date.getSeconds()+'').padStart(2,'0')` but the verboseness of this solution is preferable, definitely makes it more readable.

File: src/pages/test-report/components/timer/timer.ts [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+  }
+
+  onTouchStart(): void {
+    clearTimeout(this.touchTimeout);
+    this.touchState = true;
+  }
+
+  onTouchEnd(): void {
+    // defer the removal of the touch state to allow the page to render
+    this.touchTimeout = setTimeout(() => this.touchState = false, this.touchStateDelay);
```
Comment (buzzub) [inline src/pages/test-report/components/timer/timer.ts]: Where is `touchState` used? Is this to allow the button tap feedback?

Comment (ammarhaiderbjss) [reply src/pages/test-report/components/timer/timer.ts]: Good catch, I'll remove this

File: src/pages/test-report/components/timer/timer.scss [hunk 3]
```diff
@@ -24,7 +24,7 @@ timer {
     &.paused {
       color: map-get($colors-gds, 'gds-light-blue');
     }
-    text-align: center;
+    position: absolute;
```
Comment (buzzub) [inline src/pages/test-report/components/timer/timer.scss]: Is this to stop the text wobbling? Because a `1` character takes up less space than a `5` ?

Comment (ammarhaiderbjss) [reply src/pages/test-report/components/timer/timer.scss]: Yep, prevents the wobble/bounce

**Pull request merged**